Sources/SnapshotPreviewsCore/ExpandingViewController.swift

```diff
@@ -69,19 +76,38 @@ public final class ExpandingViewController: UIHostingController<EmergeModifierVi
     }
   }
 
-  private func runCallback() {
+  private func runCallback(_ error: Error? = nil) {
     guard !didCall else { return }
 
     didCall = true
-    expansionSettled?(rootView.emergeRenderingMode, rootView.precision, rootView.accessibilityEnabled)
+    expansionSettled?(rootView.emergeRenderingMode, rootView.precision, rootView.accessibilityEnabled, error)
+    stopAndResetTimer()
   }
 
   public override func viewDidLayoutSubviews() {
     super.viewDidLayoutSubviews()
-    self.updateScrollViewHeight()
+
+    DispatchQueue.main.async {
+      self.updateScrollViewHeight()
+    }
   }
 
   public func updateScrollViewHeight() {
+    // Timeout limit
+    if timer == nil {
+      startTimer()
+    } else if elapsedTime >= HeightExpansionTimeLimitInSeconds {
```

Comment:
Why do we have this here and when the timer fires? Shouldn't we only need it when the timer fires? If we do need it in both places, can we pull out the common handling code into a function + add a comment about why it's called from both places?

Reply:
yeah I was on the fence about this. I will remove this so it's only in the location of the timer callback